funcx_endpoint/funcx_endpoint/executors/high_throughput/executor.py

```diff
@@ -781,6 +800,67 @@ def shutdown(self, hub=True, targets='all', block=False):
         logger.info("Finished HighThroughputExecutor shutdown attempt")
         return True
 
+    def _cancel(self, future):
+        """ Attempt cancelling a task tracked by the future by requesting
+        cancellation from the interchange. Task cancellation is attempted
+        only if the future is cancellable i.e not already in a terminal
+        state. This relies on the executor not setting the task to a running
+        state, and the task only tracking pending, and completed states.
+
+        Parameters
+        ----------
+        future
+
+        Returns
+        -------
+        Bool
+        """
+
+        ret_value = future._cancel()
+        logger.debug("Sending cancel of task_id:{future.task_id} to interchange")
+        if ret_value is True:
+            self.command_client.run(TaskCancel(future.task_id))
+            logger.debug("Sent TaskCancel to interchange")
+        return ret_value
+
+
+CANCELLED = 'CANCELLED'
+CANCELLED_AND_NOTIFIED = 'CANCELLED_AND_NOTIFIED'
+FINISHED = 'FINISHED'
+
+
+class HTEXFuture(Future):
+
+    def __init__(self, executor):
+
+        super().__init__()
+        self.executor = executor
+
+    def cancel(self):
+        raise NotImplementedError(f"{self.__class__} does not implement cancel() try using best_effort_cancel()")
+
+    def _cancel(self):
+        """ Should be invoked only by the executor
+        Returns
+        -------
+        Bool
+        """
+        return super().cancel()
+
+    def best_effort_cancel(self):
+        """ Attempt to cancel the function. If the function has finished running, the task cannot be cancelled
+        and the method will return False. If the function is yet to start or is running, cancellation will be
+        attempted without guarantees, and the method will return True.
+
+        Please note that a return value of True does not guarantee that your function will not
+        execute at all, but it does guarantee that the future will be in a cancelled state.
+
```

Comment:
maybe useful to note (here or elsewhere?) there that `best_effort_cancel` does not wait until a task has been cancelled before putting the future in a cancelled state and returning: that is, a user can observe that the future says it is cancelled on the submit side, while the corresponding user code is actually still running.